redbot/cogs/mod/settings.py

```diff
@@ -283,33 +296,97 @@ async def mentionspam_ban(self, ctx: commands.Context, max_mentions: int):
 
     @modset.command()
     @commands.guild_only()
-    async def deleterepeats(self, ctx: commands.Context, repeats: int = None):
+    async def deleterepeats(self, ctx: commands.Context, repeats: int = None, *channels):
         """Enable auto-deletion of repeated messages.
 
         Must be between 2 and 20.
 
         Set to -1 to disable this feature.
+
+        Add channels if you want to custom the value for the channels (#name_of_channel)
+
+        Set to 0 if you want to use the default value for the channel selected
+
         """
+        test = 1
         guild = ctx.guild
+        # Create cache if not created
+        guild_cache = self.cache.get(guild.id, None)
+        if guild_cache is None:
+            guild_cache = self.cache[guild.id] = await create_new_cache(self.config, guild)
+
+        fails = set()
+        success = set()
+        for channel_id in channels:
+            # Removing every non-digit number
+            channel_id_digits = "".join(ch for ch in channel_id if ch.isdigit())
+            channel_id_digits
+            channel = ctx.guild.get_channel(
+                int(channel_id_digits) if len(channel_id_digits) != 0 else 0
+            )
+            if channel is None:
+                fails.add(channel_id)
+            else:
+                success.add(channel)
+
+        if len(fails) > 0:
+            msg = _("Discord failed to recognize the following channels : (")
+            for channel in fails:
+                msg += _(f"{channel},")
+            await ctx.send(msg + ")")
+
         if repeats is not None:
             if repeats == -1:
-                await self.config.guild(guild).delete_repeats.set(repeats)
-                self.cache.pop(guild.id, None)  # remove cache with old repeat limits
-                await ctx.send(_("Repeated messages will be ignored."))
+                if len(channels) != 0:
+                    channels_string = ""
+                    for channel in success:
+                        channels_string += _("{name}, ").format(name=channel.name)
+                        await self.config.channel(channel).delete_repeats.set(repeats)
+                    msg = _("Repeated messages will be ignored for the following channels : ( ")
+
+                    await ctx.send(_(msg + channels_string + ")"))
+                    self.cache[guild.id][channel.id] = defaultdict(lambda: deque(maxlen=0))
```

Comment:
```suggestion
                    channels_string = ", ".join(channel.name for channel in success)
                    for channel in success:
                        await self.config.channel(channel).delete_repeats.set(repeats)
                        self.cache[guild.id][channel.id] = defaultdict(lambda: deque(maxlen=0))

                    msg = _("Repeated messages will be ignored for the following channels: {channels_string}").format(channels_string=channels_string)
                    await ctx.send(msg)
```
You were only updating the cache for the last guild currently, since it was setting outside the loop.

Translation strings (`_("Text to translate here")`) should only be used on hardcoded strings or strings with `.format` applied **outside** of the `_()`, **not** on strings that are concatenated from multiple variables, and can thus be dynamic at runtime (IE `_(msg + channels_string + ")")` should not be a translated string).

Note that I kept `success` as the variable to loop over, this variable will need to be changed to `channels` in all places you're using it once you change the parameter to convert to channels automatically. I just don't want to add suggestions to every usage of it, since there's quite a few :P